Tunnel.js

```diff
@@ -276,22 +312,7 @@ Tunnel.prototype = util.mixin(Object.create(_super), /** @lends module:digdug/Tu
 			}
 
 			var request = sendRequest(self.url, { proxy: self.proxy });
-			request.then(
-				function (response) {
-					var decompressor = new Decompress();
-					decompressor.src(response.data)
-						.use(Decompress.zip())
-						.use(Decompress.targz())
-						.dest(self.directory)
-						.run(function (error) {
-							if (error) {
-								reject(error);
-							}
-							else {
-								resolve();
-							}
-						});
-				},
+			request.then((self.dontExtract ? writeFile : decompressFile),
```

Comment:
After looking at this a bit more, something like the following may be a cleaner solution:

```js
download: function (forceDownload, options) {
    // ...
    return new Promise(function (resolve, reject, progress, setCanceler) {
        // ...
        request.then(
            function (response) {
                return self._postDownload(response, options);
            },
            function (error) {
                // ...
            },
            progress
        );

        return request;
}	
```

Basically, leave the `download` function mostly intact, but allow an `options` argument to override `proxy`, `url`, and `isDownloaded` (it can also contain the config for `SeleniumTunnel`), and replace the decompressor code with a `_postDownload` method that can be overridden by subclasses.